src/vega.jl

```diff
@@ -1,21 +1,71 @@
-)
-
-function Vega()
-    local wavelength, flux
-    HDF5.h5open(datadep"vega/vega.hd5") do fh
-        node = fh["spectrum"]
-        data = read(node)
-        wlunit = parse_unit(read(HDF5.attributes(node)["WAVELENGTH_UNIT"]))
-        wavelength = map(i -> uconvert(wave_unit, i.WAVELENGTH * wlunit), data)
-        fluxunit = u"erg/s/cm^2/angstrom"
-        flux = map(i -> i.FLUX * fluxunit, data)
+"""
+    Vega(name::String = "alpha_lyr_stis_011")
+Loads the reference spectrum with filename `name` and returns an appropriate instance of `Vega` that can be used to compute zeropoints and magnitudes in the Vega magnitude system.
+
+If the provided `name` is the full path to the existing file on disk, the spectrum is loaded from that file. Otherwise, it is downloaded from the [CALSPEC database](https://www.stsci.edu/hst/instrumentation/reference-data-for-calibration-and-tools/astronomical-catalogs/calspec) of standard stars maintained by STScI. Files downloaded this way are cached for future use. Specifically, we draw from the extended catalog [here](https://ssb.stsci.edu/cdbs/calspec/). The CALSPEC catalog with the most recent reference spectrum for each star is located [here](https://archive.stsci.edu/hlsps/reference-atlases/cdbs/current_calspec/). Standard Vega spectra start with `"alpha_lyr"`. `name`s like `"alpha_lyr_stis_XXX"` are based on composites of calibrated stellar models and HST/STIS data, while `"alpha_lyr_mod_XXX"` are based on stellar models only.
+
+Sometimes Vega is not used as the standard star for photometric systems even when the system follows the Vega magnitude convention. For example, in the near-IR Sirius is often used as the standard reference spectrum rather than Vega. This is the case for the definition of the [JWST zeropoints](https://jwst-docs.stsci.edu/jwst-near-infrared-camera/nircam-performance/nircam-absolute-flux-calibration-and-zeropoints#gsc.tab=0), which presently use the `"sirius_stis_005.fits"` CALSPEC spectrum as their standard. To load a different standard, simply provide the corresponding `name` to `Vega`. For example, to load the Sirius spectrum, use `Vega("sirius_stis_005")` (the `".fits"` extension is optional).
+"""
+function Vega(name::String = "alpha_lyr_stis_011")
+    name = splitext(name)[1] * ".fits" # ensure .fits extension
+    if isfile(name)
+        return FITSIO.FITS(name, "r") do f
+            wavelength = read(f[2], "WAVELENGTH") .* u"angstrom"
+            flux = read(f[2], "FLUX") .* u"erg/s/cm^2/angstrom"
+            Vega(wavelength, flux, basename(name))
+        end
     end
-    return Vega(wavelength, flux, "vega.hdf5")
+
+    # Load from CALSPEC database or local cache
+    fname = joinpath(vega_cache, name)
+    if !isfile(fname)
+        HTTP.download("https://ssb.stsci.edu/cdbs/calspec/" * name, fname)
+    end
+    return Vega(fname)
+end
+Vega(name::AbstractString) = Vega(string(name))
+
+"""
+    get_calspec_names([substring::AbstractString])
```

Comment:
It may be worth adding somewhere here that this function is not exported. From the example below, it would look like `Vega` and `get_calspec_names` are treated the same, but actually only `Vega` is exported.